src/main/java/ca/nrc/cadc/web/RestletPrincipalExtractor.java

```diff
@@ -107,80 +168,24 @@ public X509CertificateChain getCertificateChain()
 
         final Set<Principal> principals = new HashSet<>();
 
-        addHTTPPrincipal(principals);
+        // For now, the UI only needs to deal with the cookie principal.
+        addCookiePrincipal(principals);
 
         return principals;
     }
 
-    @Override
-    public DelegationToken getDelegationToken()
-    {
-        return null;
-    }
-
-    /**
-     * Add the HTTP Principal, if it exists.
-     */
-    private void addHTTPPrincipal(final Set<Principal> principals)
+    private void addCookiePrincipal(final Set<Principal> principals)
     {
         init();
 
-        final String httpUser = getAuthenticatedUsername();
-
-        // only add one HttpPrincipal, precedence order
-        if (StringUtil.hasText(httpUser)) // user from HTTP AUTH
-        {
-            principals.add(new HttpPrincipal(httpUser));
-        }
-
-        if (cookiePrincipal != null) // user from cookie
-        {
-            principals.add(cookiePrincipal);
-        }
-    }
-
-
-    /**
-     * Obtain the Username submitted with the Request.
-     *
-     * @return String username, or null if none found.
-     */
-    private String getAuthenticatedUsername()
-    {
-        final String username;
-
-        if (!getRequest().getClientInfo().getPrincipals().isEmpty())
-        {
-            // Put in to support Safari not injecting a Challenge Response.
-            // Grab the first principal's name as the username.
-            // update: this is *always* right and works with realms; the previous
-            // call to getRequest().getChallengeResponse().getIdentifier() would
-            // return whatever username the caller provided in a non-authenticating call
-            username = getRequest().getClientInfo().getPrincipals().get(0)
-                    .getName();
-        }
-        else
+        if (cookieCredential != null)
         {
-            username = null;
+            principals.add(new CookiePrincipal("CADC_SSO", cookieCredential.getSsoCookieValue()));
```

Comment:
Use `SSOCookieManager.DEFAULT_SSO_COOKIE_NAME`.  If that cookie name ever changes this'll kick us a bit.

Reply:
Done. :)